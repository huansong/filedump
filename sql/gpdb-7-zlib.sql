-- 1. Tests for pre-generated relation files

-- create table <name> (c1 int, c2 text)
--   with (appendonly=true, compresstype=zlib)
\! pg_filedump -z -M -k -T zlib test/ao.zlib.gp7

-- create table <name> (c1 int, c2 text)
--   with (appendonly=true, orientation=column, compresstype=zlib)
\! pg_filedump -z -M -k -O column -T zlib test/co.zlib.gp7

-- 2. Tests for new relations files generated by the running GPDB server

create table ao_zlib(c1 int, c2 text) with (appendonly=true, compresstype=zlib) distributed replicated;
create table co_zlib(c1 int, c2 text) with (appendonly=true, orientation=column, compresstype=zlib) distributed replicated;
insert into ao_zlib select i,'abc' from generate_series(1,10)i;
insert into co_zlib select i,'abc' from generate_series(1,10)i;
\set relname ao_zlib
\set ext '.1'
\setenv opt '-z -M -k -T zlib'
\ir run_test.sql
\set relname co_zlib
\set ext '.1'
\setenv opt '-z -M -k -O column -T zlib'
\ir run_test.sql

