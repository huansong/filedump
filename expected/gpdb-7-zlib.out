-- 1. Tests for pre-generated relation files
-- create table <name> (c1 int, c2 text)
--   with (appendonly=true, compresstype=zlib)
\! pg_filedump -z -M -k -T zlib test/ao.zlib.gp7

*******************************************************************
* PostgreSQL File/Block Formatted Dump Utility
*
* File: test/ao.zlib.gp7
* Options used: -z -M -k -T zlib
*******************************************************************

 Initial Bytes Read = 624

***** Block    0 Start ***( Position:    0) *************************
<AO Header> -----
 HeaderKind: 1 
 Header Checksum Valid (checksum 0x966160f8)
(AoHeaderKind_SmallContent)
 executorBlockKind: 1 rowCount: 38
 headerLen: 24 uncompressedLen: 692 compressedLen: 282
 firstRowNumber: 1
Content checksum valid (checksum: 0xb36507cb)

***** Block    0 End (Total Block length:  312) ******************
 Initial Bytes Read = 312

***** Block    1 Start ***( Position:  312) *************************
<AO Header> -----
 HeaderKind: 1 
 Header Checksum Valid (checksum 0x1a359a99)
(AoHeaderKind_SmallContent)
 executorBlockKind: 1 rowCount: 38
 headerLen: 24 uncompressedLen: 692 compressedLen: 282
 firstRowNumber: 101
Content checksum valid (checksum: 0x238fe127)

***** Block    1 End (Total Block length:  312) ******************
 Initial Bytes Read = 0

 Done DumpAppendOnlyFileContents.....
-- create table <name> (c1 int, c2 text)
--   with (appendonly=true, orientation=column, compresstype=zlib)
\! pg_filedump -z -M -k -O column -T zlib test/co.zlib.gp7

*******************************************************************
* PostgreSQL File/Block Formatted Dump Utility
*
* File: test/co.zlib.gp7
* Options used: -z -M -k -O column -T zlib
*******************************************************************

 Initial Bytes Read = 240

***** Block    0 Start ***( Position:    0) *************************
<AO Header> -----
 HeaderKind: 1 
 Header Checksum Valid (checksum 0x368292ac)
(AoHeaderKind_SmallContent)
 executorBlockKind: 1 rowCount: 38
 headerLen: 24 uncompressedLen: 168 compressedLen: 91
 firstRowNumber: 1
Content checksum valid (checksum: 0xf2163d87)

	<AO Content Info> -----
	 Decompressing using "ZLIB" with compressionLevel: 0
	 DatumStreamVersion_Original  flags: 0
	 ndatum: 38 nullSize: 0 Size: 152
***** Block    0 End (Total Block length:  120) ******************
 Initial Bytes Read = 120

***** Block    1 Start ***( Position:  120) *************************
<AO Header> -----
 HeaderKind: 1 
 Header Checksum Valid (checksum 0xd73b437d)
(AoHeaderKind_SmallContent)
 executorBlockKind: 1 rowCount: 38
 headerLen: 24 uncompressedLen: 168 compressedLen: 91
 firstRowNumber: 101
Content checksum valid (checksum: 0x914c4e63)

	<AO Content Info> -----
	 Decompressing using "ZLIB" with compressionLevel: 0
	 DatumStreamVersion_Original  flags: 0
	 ndatum: 38 nullSize: 0 Size: 152
***** Block    1 End (Total Block length:  120) ******************
 Initial Bytes Read = 0

 Done DumpAppendOnlyFileContents.....
-- 2. Tests for new relations files generated by the running GPDB server
create table ao_zlib(c1 int, c2 text) with (appendonly=true, compresstype=zlib) distributed replicated;
create table co_zlib(c1 int, c2 text) with (appendonly=true, orientation=column, compresstype=zlib) distributed replicated;
insert into ao_zlib select i,'abc' from generate_series(1,10)i;
insert into co_zlib select i,'abc' from generate_series(1,10)i;
\set relname ao_zlib
\set ext '.1'
\setenv opt '-z -M -k -T zlib'
\ir run_test.sql
\echo Testing :relname
Testing ao_zlib
checkpoint;
select setting as datadir from gp_settings where gp_segment_id = 0 and name = 'data_directory' \gset
select oid as db_oid from pg_database where datname = current_database() \gset
select relfilenode from gp_dist_random('pg_class') where gp_segment_id = 0 and relname = :'relname' \gset
\set filepath :datadir '/base/' :db_oid '/' :relfilenode :ext
\setenv filepath :filepath
\! pg_filedump $opt $filepath | sed -e "s/logid      ./logid      ./" -e "s/recoff 0x......../recoff 0x......../" -e "s/File: .*/File: /" -e "s/HeapId \(.*\)/HeapId \(\)/" -e "s/IndexId \(.*\)/IndexId \(\)/" -e "s/Checksum: ....../Checksum: /"

*******************************************************************
* PostgreSQL File/Block Formatted Dump Utility
*
* File: 
* Options used: -z -M -k -T zlib
*******************************************************************

 Initial Bytes Read = 104

***** Block    0 Start ***( Position:    0) *************************
<AO Header> -----
 HeaderKind: 1 
 Header Checksum Valid (checksum 0x50311553)
(AoHeaderKind_SmallContent)
 executorBlockKind: 1 rowCount: 10
 headerLen: 24 uncompressedLen: 188 compressedLen: 78
 firstRowNumber: 1
Content checksum valid (checksum: 0x42b1abb5)

***** Block    0 End (Total Block length:  104) ******************
 Initial Bytes Read = 0

 Done DumpAppendOnlyFileContents.....
--
----------------------------------------------------------------------------------------------
--
\set relname co_zlib
\set ext '.1'
\setenv opt '-z -M -k -O column -T zlib'
\ir run_test.sql
\echo Testing :relname
Testing co_zlib
checkpoint;
select setting as datadir from gp_settings where gp_segment_id = 0 and name = 'data_directory' \gset
select oid as db_oid from pg_database where datname = current_database() \gset
select relfilenode from gp_dist_random('pg_class') where gp_segment_id = 0 and relname = :'relname' \gset
\set filepath :datadir '/base/' :db_oid '/' :relfilenode :ext
\setenv filepath :filepath
\! pg_filedump $opt $filepath | sed -e "s/logid      ./logid      ./" -e "s/recoff 0x......../recoff 0x......../" -e "s/File: .*/File: /" -e "s/HeapId \(.*\)/HeapId \(\)/" -e "s/IndexId \(.*\)/IndexId \(\)/" -e "s/Checksum: ....../Checksum: /"

*******************************************************************
* PostgreSQL File/Block Formatted Dump Utility
*
* File: 
* Options used: -z -M -k -O column -T zlib
*******************************************************************

 Initial Bytes Read = 64

***** Block    0 Start ***( Position:    0) *************************
<AO Header> -----
 HeaderKind: 1 
 Header Checksum Valid (checksum 0xa809e79b)
(AoHeaderKind_SmallContent)
 executorBlockKind: 1 rowCount: 10
 headerLen: 24 uncompressedLen: 56 compressedLen: 39
 firstRowNumber: 1
Content checksum valid (checksum: 0x6ec3fc13)

	<AO Content Info> -----
	 Decompressing using "ZLIB" with compressionLevel: 0
	 DatumStreamVersion_Original  flags: 0
	 ndatum: 10 nullSize: 0 Size: 40
***** Block    0 End (Total Block length:   64) ******************
 Initial Bytes Read = 0

 Done DumpAppendOnlyFileContents.....
--
----------------------------------------------------------------------------------------------
--
